presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftSessionProperties.java

```diff
@@ -27,15 +28,26 @@
 public final class ThriftSessionProperties
 {
     private final List<PropertyMetadata<?>> sessionProperties;
+    private static final String SET_THRIFT_IDENTITY_HEADER = "THRIFT_CONNECTOR_SET_IDENTITY_HEADERS";
 
     @Inject
     public ThriftSessionProperties(ThriftConnectorConfig config)
     {
-        sessionProperties = ImmutableList.of();
+        PropertyMetadata<Boolean> s1 = PropertyMetadata.booleanProperty(
```

Comment:
Note: Decision to include this as a Session Property was discussed in [this PR](https://github.com/prestodb/presto/pull/16319).